float32 lookahead_dist
float32 velocity
